unhandled exception for NoneType

Counterparty has crashed 3 times in the past 2 days with the error below. 

Restarting the server seems to get things operating normally again.

```
counterparty_1           | [2017-11-07 10:12:30][INFO] Block: 493449 (5.77s, hashes: L:4e2fc / TX:20f96 / M:d0cff)
counterparty_1           | [2017-11-07 10:12:50][WARNING] Error composing order transaction via API: Insufficient BTC at address 1AgDE6HaVpRX3KFvDBopGXCfiaqf5nzub4. (Need approximately 0.00059609 BTC.) To spend unconfirmed coins, use the flag `--unconfirmed`. (Unconfirmed coins cannot be spent from multi‐sig addresses.)
counterparty_1           | [2017-11-07 10:16:57][WARNING] Error composing send transaction via API: Insufficient BTC at address 1G1QNA3WGY8QpNrxVsSV4H3naBqbHiT4D7. (Need approximately 0.00093771 BTC.) To spend unconfirmed coins, use the flag `--unconfirmed`. (Unconfirmed coins cannot be spent from multi‐sig addresses.)
counterparty_1           | [2017-11-07 10:21:57][WARNING] tx missing in rawtx cache: 'a0d6ca59666025cbbbb9862ba30fb03f0ab3b0a831d96864ff926a58fe8105ec' -- txhash_list size: 2, hash: d8a13bede463fe0b8f94b73cb4ddda77 / raw_transactions_cache size: 19999 / # rpc_batch calls: 2 / txhash in noncached_txhashes: True / txhash in txhash_list: True -- list []
counterparty_1           | [2017-11-07 10:23:49][ERROR] Unhandled Exception
counterparty_1           | Traceback (most recent call last):
counterparty_1           |   File "/usr/local/bin/counterparty-server", line 9, in <module>
counterparty_1           |     load_entry_point('counterparty-cli', 'console_scripts', 'counterparty-server')()
counterparty_1           |   File "/counterparty-cli/counterpartycli/__init__.py", line 16, in server_main
counterparty_1           |     server.main()
counterparty_1           |   File "/counterparty-cli/counterpartycli/server.py", line 157, in main
counterparty_1           |     server.start_all(db)
counterparty_1           |   File "/counterparty-lib/counterpartylib/server.py", line 408, in start_all
counterparty_1           |     blocks.follow(db)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 1299, in follow
counterparty_1           |     mempool_tx_index = list_tx(db, None, block_index, curr_time, tx_hash, tx_index=mempool_tx_index, tx_hex=tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 912, in list_tx
counterparty_1           |     source, destination, btc_amount, fee, data = get_tx_info(tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 450, in get_tx_info
counterparty_1           |     return _get_tx_info(tx_hex, block_parser, block_index)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 460, in _get_tx_info
counterparty_1           |     return  get_tx_info3(tx_hex, block_parser=block_parser)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 583, in get_tx_info3
counterparty_1           |     return get_tx_info2(tx_hex, block_parser=block_parser, p2sh_support=True)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 591, in get_tx_info2
counterparty_1           |     ctx = backend.deserialize(tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/backend/__init__.py", line 83, in deserialize
counterparty_1           |     return bitcoinlib.core.CTransaction.deserialize(binascii.unhexlify(tx_hex))
counterparty_1           | TypeError: argument should be bytes, buffer or ASCII string, not 'NoneType'
federatednode_counterparty_1 exited with code 0

Issue just happened again.

```counterparty_1           | [2017-11-07 21:35:06][INFO] Block: 493529 (5.73s, hashes: L:e5860 / TX:5f9e2 / M:3bcf0)
counterparty_1           | [2017-11-07 21:36:47][WARNING] Error composing send transaction via API: Insufficient BTC at address 17spyhNqANBNiZfyRxV4eVHoZhNPYxn8ST. (Need approximately 0.00125429 BTC.) To spend unconfirmed coins, use the flag `--unconfirmed`. (Unconfirmed coins cannot be spent from multi‐sig addresses.)
counterparty_1           | [2017-11-07 21:38:48][ERROR] Unhandled Exception
counterparty_1           | Traceback (most recent call last):
counterparty_1           |   File "/usr/local/bin/counterparty-server", line 9, in <module>
counterparty_1           |     load_entry_point('counterparty-cli', 'console_scripts', 'counterparty-server')()
counterparty_1           |   File "/counterparty-cli/counterpartycli/__init__.py", line 16, in server_main
counterparty_1           |     server.main()
counterparty_1           |   File "/counterparty-cli/counterpartycli/server.py", line 157, in main
counterparty_1           |     server.start_all(db)
counterparty_1           |   File "/counterparty-lib/counterpartylib/server.py", line 408, in start_all
counterparty_1           |     blocks.follow(db)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 1299, in follow
counterparty_1           |     mempool_tx_index = list_tx(db, None, block_index, curr_time, tx_hash, tx_index=mempool_tx_index, tx_hex=tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 912, in list_tx
counterparty_1           |     source, destination, btc_amount, fee, data = get_tx_info(tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 450, in get_tx_info
counterparty_1           |     return _get_tx_info(tx_hex, block_parser, block_index)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 460, in _get_tx_info
counterparty_1           |     return  get_tx_info3(tx_hex, block_parser=block_parser)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 583, in get_tx_info3
counterparty_1           |     return get_tx_info2(tx_hex, block_parser=block_parser, p2sh_support=True)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/blocks.py", line 591, in get_tx_info2
counterparty_1           |     ctx = backend.deserialize(tx_hex)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/backend/__init__.py", line 83, in deserialize
counterparty_1           |     return bitcoinlib.core.CTransaction.deserialize(binascii.unhexlify(tx_hex))
counterparty_1           | TypeError: argument should be bytes, buffer or ASCII string, not 'NoneType'
federatednode_counterparty_1 exited with code 0
```

--- a/counterpartylib/lib/blocks.py
+++ b/counterpartylib/lib/blocks.py
@@ -1296,6 +1296,9 @@ def follow(db):
                                       )
 
                         tx_hex = raw_transactions[tx_hash]
+                        if tx_hex is None:
+                          logger.debug('tx_hash %s not found in backend.  Not adding to mempool.', (tx_hash, ))
+                          raise MempoolError
                         mempool_tx_index = list_tx(db, None, block_index, curr_time, tx_hash, tx_index=mempool_tx_index, tx_hex=tx_hex)
 
                         # Parse transaction.

